### DISCUSSION TITLE
Download all starred videos?

### DISCUSSION THREAD

**Discussion by thinkloop** [blakeblackshear/frigate#13445] [Ask A Question] [answered]
### Describe the problem you are having

The 0.14 release notes say to "export any starred/favorite events before upgrading". Is there a way to do that without clicking each one individually?

### Version

0.13

### Frigate config file

```yaml
n/a
```

### Relevant Frigate log output

```shell
n/a
```

### Relevant go2rtc log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Frigate stats

```json
n/a
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

n/a

**Comment by NickM-27** [comment]
you could write a script to do it, but no there is no way in the old UI to do that

**Reply by thinkloop** [reply]
Are you thinking something like a selenium script, or is there an api you had in mind?

**Reply by NickM-27** [reply]
just a simple bash script, psuedo code:

```
events = `/api/events?starred=1`

for event in events:
    `/api/export/start/{event.start_time}/end/{event.end_time}`
```

this could also be run after updating

**Reply by thinkloop** [reply]
ty

**Reply by thinkloop** [reply]
What would be a good way to wait for, or get, the results of each export? For example I ran out of space during the process but my script had no visibility into that problem: `frigate    | s6-log: warning: unable to write to /dev/shm/logs/frigate/current: No space left on device`. Is there a synchronous way of exporting?

**Reply by NickM-27** [reply]
there isn't a good way since the api call doesn't wait for completion to respond as that would take way too long

**Comment by thinkloop** [comment, marked as answer]
If anyone is interested I put together a Node.js script that automates the process: https://github.com/thinkloop/frigate-star-export

It goes through each starred video and triggers an export, pausing in between based on clip duration.
